odo/backends/url.py

```diff
@@ -133,7 +133,9 @@ def append_urlX_to_X(target, source, **kwargs):
         with open(target.path, 'wb') as fp:
             for chunk in iter(curry(r.read, chunk_size), ''):
                 fp.write(chunk)
-                return target
+                if not chunk:
```

Comment:
Do we ever hit this branch? The second argument to iter here should prevent us from entering the body when we read an empty string, for example:

```python
In [13]: def f(it=iter(['a', 'b', 'c', ''])):
   ....:     return next(it)
   ....: 

In [14]: tuple(iter(f, ''))
Out[14]: ('a', 'b', 'c')
```

One thing we may want to do though is change this loop to break when `len(chunk) < chunk_size` because this will avoid that last empty read, but idk if that is worth it.